model/lifecycle.go

```diff
@@ -845,9 +851,26 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
 	return nil
 }
 
+func getTaskCreateTime(project *Project, v *version.Version) (*time.Time, error) {
+	if evergreen.IsPatchRequester(v.Requester) {
+		baseVersion, err := version.FindOne(version.BaseVersionFromPatch(project.Identifier, v.Revision))
+		if err != nil {
+			grip.Error(errors.Wrap(err, "Error finding base version for patch version"))
+			return nil, err
+		} else if baseVersion == nil {
```

Comment:
it's possible for these db functions to return nil errors and nil data, so I think it's better to not use an else.

Reply:
I'm not sure I understand what you mean.
The check in the "else if" is handling the case you mention of nil error and nil data.
Also the if clause returns, so in this case the behavior is the same with or without the else clause.

Reply:
The else if doesn't run if err is nil